spacy/gold.pyx

```diff
@@ -212,6 +212,7 @@ class GoldCorpus(object):
                 if self.limit and i >= self.limit:
                     break
                 i += 1
+            paragraph_tuples.append(cats) # restore original data
```

Comment:
This isn't needed, is it?

Reply:
Right now it isn't, no, because `self.train_tuples` is implemented as a generator. However if this would ever be implemented different, e.g. a simple field holding a static `list`, than this `count_train()` function would be removing data. So for robustness and bug avoidance I think it makes sense to restore the data...